src/app/shared/top-panel/search-box/search-box.component.ts

```diff
@@ -90,12 +90,14 @@ export class SearchBoxComponent implements OnInit, OnDestroy {
   }
 
   private subscribeToQuery() {
-    const querySubscription = combineLatest(this.store$.pipe(select(selectQuery)), this.loadData(), this.router.events)
+    const querySubscription = combineLatest(this.store$.pipe(select(selectQuery)), this.loadData())
       .pipe(
         debounceTime(100),
-        filter(([query, , event]) => !!query && event instanceof NavigationEnd),
-        tap(([, data]) => (this.queryData = data)),
-        map(([query, data]) => ({queryItems: new QueryItemsConverter(data).fromQuery(query, true), query})),
+        withLatestFrom(this.router.events.pipe(startWith(null))),
```

Comment:
Why are router events needed here?

Reply:
The reason is that `search-box.component` is always visible on the page and sometimes during navigation this observable emits value before query changed (because data are cleaned during navigation) and then it tries to override query which leads to unexpected state. I need to be sure that observable emits data only when `NavigationEnd` event is the latest one.

But let me test this once again just to make sure everything works properly!